api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -146,19 +205,21 @@ private Order saveOrder(Order order, OrderContext orderContext, boolean isRetros
 		if (isDrugOrder) {
 			((DrugOrder) order).setAutoExpireDateBasedOnDuration();
 		}
-		
 		if (concept == null) {
 			throw new MissingRequiredPropertyException("Order.concept.required");
 		}
-		
+
+	}
+
+	private void ensureOrderTypeIsSet(Order order, OrderContext orderContext) {
 		Order previousOrder = order.getPreviousOrder();
 		if (order.getOrderType() == null) {
 			OrderType orderType = null;
 			if (orderContext != null) {
 				orderType = orderContext.getOrderType();
 			}
 			if (orderType == null) {
-				orderType = getOrderTypeByConcept(concept);
+				orderType = getOrderTypeByConcept(order.getConcept());
 			}
 			//Check if it is instance of DrugOrder
```

Comment:
remove all comments in this method, we see from the code that this is what is happening